pages/sign-in/kakao/index.page.tsx

```diff
@@ -1,18 +1,39 @@
-import { GetServerSidePropsContext } from "next";
+import { GetServerSidePropsContext, GetServerSidePropsResult } from "next";
+import { useEffect } from "react";
+
+import { useRouter } from "next/router";
 
 import { instance } from "@/apis/axios";
 
-export async function getServerSideProps(context: GetServerSidePropsContext) {
+interface KakaoProps {
+  isFirst: boolean;
+}
+
+export async function getServerSideProps(
+  context: GetServerSidePropsContext,
+): Promise<GetServerSidePropsResult<KakaoProps>> {
   const code = context.query["code"];
 
   const res = await instance.get(`/login/oauth/kakao?code=${code}`);
-  const loginData = res.data;
+  const { isFirst } = res.data;
 
   return {
-    props: {},
+    props: { isFirst: Boolean(isFirst) },
   };
 }
 
-export default function Kakao() {
+export default function Kakao({ isFirst }: KakaoProps) {
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isFirst) {
+      router.push("/profile-setup"); // TODO: 우혁님 PR merge 된 후에, 상수로 수정 예정
+    }
+
+    if (!isFirst) {
+      router.push("/");
+    }
```

Comment:
```suggestion
    if (!isFirst) {
      router.push("/");
    }
```
`boolean`값 이라서 `true` or `false`인 것 같은데 한번 더 if문으로 감싸신 이유가 있으신가요??

Reply:
```tsx
isFirst ? router.push("/profile-setup") : router.push("/") 
```

이런 식으로 작성하는 것도 가능하긴 하지만, 개인적으로는 `if` 문을 사용해주는게 가독성이 더 좋아보인다고 생각해서 저렇게 작성했었습니다🐸 `if`문은 사용하지 않는게 더 나을까요?

Reply:
저도 if문 사용하는건 좋은 것 같은데 지금 남주님 코드는 if문을 2번 사용하셨는데 저는 if문 한번 사용할 때 early return을 사용해서 아래 if문은 생략하는건 어떤지 여쭤본거였습니다!
```js
if (isFirst) {
   router.push("/profile-setup"); 
   return; //early return
}

router.push("/"); // if문 없이 코드 작성
```

Reply:
아 early return 사용하는거 좋은 것 같아요 수정해보겠습니다!